### ISSUE TITLE
armpolicy - Error when unmarshalling PolicyAssignment

### ISSUE THREAD

**Issue by sthordall** [Azure/azure-sdk-for-go#16734]
# Bug Report

Module: `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armpolicy`
Version: `v0.1.1`
SDK Version: `v61.0.0`
Go Version output:  `go version go1.17.5 linux/amd64` 

## What happened?
Unmarshalling errors when requesting PolicyAssignments, via `PolicyAssignmentsClient->ListForManagementGroup`

Both for strings and arrays in parameters, which are both standard for policy assignments.
```
unmarshalling type *armpolicy.PolicyAssignmentListResult: json: cannot unmarshal string into Go struct field ParameterValuesValue.value.properties.parameters.value of type map[string]interface {}
```

```
unmarshalling type *armpolicy.PolicyAssignmentListResult: json: cannot unmarshal array into Go struct field ParameterValuesValue.value.properties.parameters.value of type map[string]interface {}
```

The response format from the Azure API is as follows on Policy Assignments (removed `metadata` and `systemdata`):
```json
{
            "properties": {
                "displayName": "Not allowed resource types - Virtual Machine",
                "policyDefinitionId": "/providers/Microsoft.Authorization/policyDefinitions/6c112d4e-5bc7-47ae-a041-ea2d9dccd749",
                "scope": "/providers/Microsoft.Management/managementGroups/12614148-c8b0-4bc7-a0c0-594f14bd0864",
                "parameters": {
                    "effect": {
                        "value": "Audit"
                    },
                    "listOfResourceTypesNotAllowed": {
                        "value": [
                            "Microsoft.Compute/virtualMachines"
                        ]
                    }
                },
                "description": "Denies creation of Virtual Machines under the assigned scope",
                "metadata": {
                },
                "enforcementMode": "Default"
            },
            "id": "/providers/Microsoft.Management/managementGroups/12614148-c8b0-4bc7-a0c0-594f14bd0864/providers/Microsoft.Authorization/policyAssignments/no_vm",
            "type": "Microsoft.Authorization/policyAssignments",
            "name": "no_vm",
            "systemData": {
            }
        },
```

Whereas the `PolicyAssignmentProperties` type in the SDK expects `map[string]interface{}` in the parameters `value` fields, i.e. `ParameterValuesValue`:
```go
// PolicyAssignmentProperties - The policy assignment properties.
type PolicyAssignmentProperties struct {
	// This message will be part of response in case of policy violation.
	Description *string `json:"description,omitempty"`

	// The display name of the policy assignment.
	DisplayName *string `json:"displayName,omitempty"`

	// The policy assignment enforcement mode. Possible values are Default and DoNotEnforce.
	EnforcementMode *EnforcementMode `json:"enforcementMode,omitempty"`

	// The policy assignment metadata. Metadata is an open ended object and is typically a collection of key value pairs.
	Metadata map[string]interface{} `json:"metadata,omitempty"`

	// The messages that describe why a resource is non-compliant with the policy.
	NonComplianceMessages []*NonComplianceMessage `json:"nonComplianceMessages,omitempty"`

	// The policy's excluded scopes.
	NotScopes []*string `json:"notScopes,omitempty"`

	// The parameter values for the assigned policy rule. The keys are the parameter names.
	Parameters map[string]*ParameterValuesValue `json:"parameters,omitempty"`

	// The ID of the policy definition or policy set definition being assigned.
	PolicyDefinitionID *string `json:"policyDefinitionId,omitempty"`

	// READ-ONLY; The scope for the policy assignment.
	Scope *string `json:"scope,omitempty" azure:"ro"`
}

// ParameterValuesValue - The value of a parameter.
type ParameterValuesValue struct {
	// The value of the parameter.
	Value map[string]interface{} `json:"value,omitempty"`
}
```

## What did you expect or want to happen?
No errors and successful unmarshalling of json. 

## How can we reproduce it?
Run the following against a subscription with policy assignments on a management group. Policy assignment should contain parameters, such as `effect` or `listOfResourceTypesNotAllowed`.
 
```go
cred, err := azidentity.NewDefaultAzureCredential(nil)
ctx := context.Background()
client := armpolicy.NewPolicyAssignmentsClient("<subscription-id>", cred, nil)
pager := client.ListForManagementGroup("<management-group-id>", &armpolicy.PolicyAssignmentsListForManagementGroupOptions{Filter: to.StringPtr("atExactScope()"), Top: nil, }) 

pager.NextPage(ctx)
if err := pager.Err(); err != nil {
    log.Fatalf("failed to advance page: %v", err)
}  
```

## Input
Change `ParameterValuesValue` to:
```go
type ParameterValuesValue struct {
	// The value of the parameter.
	Value interface{} `json:"value,omitempty"`
}
```

I'd happily help out, but I'm not sure where to start since this seems to be auto-generated.

I've experienced the same kind of error with `PolicyDefinition` and I can see #16715 is related, so it seems to be a general issue with the auto-generation of some response structs.

Thanks,
Stephan

**Comment by jhendrixMSFT** [comment]
This looks to be an issue with the service swagger.  The [spec](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/policyAssignments.json#L800) states that `value` is a JSON object which is clearly not the case.  This is likely being uncovered now due to strict conformance made in the updated modeler some time last year.

**Comment by kenieva** [comment]
Hi @sthordall, thanks for bringing this up to our attention. I do have some clarifying questions? What representation are you trying to convert to? Also, does this work on other areas of Policy SDK?

**Comment by sthordall** [comment]
Hi @kenieva. 

Can you clarify what you mean with representation conversion? I experience the error, just by calling the SDK.

I've experienced the same error with Policy Definitions; I haven't tried the rest.

**Comment by tadelesh** [comment]
@sthordall We've released the 0.3.0 version of `armpolicy` which has resolved this issue. Could you please upgrade your dependency? Thanks.

**Comment by tadelesh** [comment]
@sthordall We've released the 0.3.0 version of armpolicy which has resolved this issue. Could you please upgrade your dependency? Thanks.

**Comment by sthordall** [comment]
I can confirm that it works as expected on version 0.3.0. Thanks!
